### PR TITLE
feat add login-page

### PR DESCRIPTION
**Pull request by nurba710**

### CODE REVIEW

**Review by UktamTursunov — APPROVED** [review]

File: src/components/LoginForm/ChangePassword.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import React from 'react'
+import { Formik } from 'formik'
+import { Error, FormContainer, FormTitle, InputContainer } from './Style'
+import Input from '../Input/Input'
+import { ButtonStyle } from '../Button/style'
+import * as yup from 'yup'
+import { useChangePasswordPOSTMutation } from '../../service/ChangePassword'
+import { NavLink } from 'react-router-dom'
+
+interface valuesType {
```
Comment (Ruslan1516) [inline src/components/LoginForm/ChangePassword.tsx:10]: move valuesType to types file

File: src/components/LoginForm/ChangePassword.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+const ChangePassword: React.FC = () => {
+	const validationSchema = yup.object().shape({
+		password_new: yup.string().required('Обязательное поле'),
+		password_old: yup.string().required('Обязательное поле'),
+	})
+
+	const [change] = useChangePasswordPOSTMutation()
+
+	const handleSubmit = async (values: valuesType) => {
```
Comment (Ruslan1516) [inline src/components/LoginForm/ChangePassword.tsx:23]: why do you need async in this function?

File: src/components/LoginForm/ChangePassword.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+		await change(values).unwrap()
+	}
+
+	return (
+		<Formik
+			initialValues={{
```
Comment (Ruslan1516) [inline src/components/LoginForm/ChangePassword.tsx:29]: move initialValues to consts file

File: src/components/LoginForm/ChangePassword.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+							value={values.password_new}
+							type={'password'}
+							onBlur={handleBlur}
+						/>
+						{touched.password_new && errors.password_new && <Error>{errors.password_new}</Error>}
+					</InputContainer>
+					<ButtonStyle width='120px' height='50px' disabled={!isValid && !dirty} type={'submit'}>
+						Change Password
+					</ButtonStyle>
+					<NavLink to='/success'>Вернуться на главную страницу</NavLink>
```
Comment (Ruslan1516) [inline src/components/LoginForm/ChangePassword.tsx:62]: create "/success" const

File: src/components/LoginForm/LoginForm.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+import { ButtonStyle } from '../Button/style'
+import Input from '../Input/Input'
+import { useLoginPOSTMutation } from '../../service/login'
+import { Formik } from 'formik'
+import * as yup from 'yup'
+import { getStorage, setStorage } from '../../common/services/storage.service'
+import { TOKEN } from '../../common/consts/auth.const'
+import { useNavigate } from 'react-router-dom'
+
+interface valuesType {
```
Comment (Ruslan1516) [inline src/components/LoginForm/LoginForm.tsx:12]: and this

File: src/components/LoginForm/LoginForm.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+	username: string
+	password: string
+}
+
+const LoginForm: React.FC = () => {
+	const validationSchema = yup.object().shape({
```
Comment (Ruslan1516) [inline src/components/LoginForm/LoginForm.tsx:18]: move validationSchema to other file

File: src/components/LoginForm/LoginForm.tsx [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+	useEffect(() => {
+		if (data?.data.token) setStorage(TOKEN, data?.data.token)
+	}, [data])
+
+	const token = getStorage(TOKEN)
+
+	useEffect(() => {
+		if (token) {
+			navigate('/success')
```
Comment (Ruslan1516) [inline src/components/LoginForm/LoginForm.tsx:34]: and this "success"

File: src/components/LoginForm/LoginForm.tsx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+		} else {
+			navigate('/')
+		}
+	}, [navigate, token, data])
+
+	const handleSubmit = async (values: valuesType) => {
```
Comment (Ruslan1516) [inline src/components/LoginForm/LoginForm.tsx:40]: why do you need async in this function?

File: src/components/LoginForm/LoginForm.tsx [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+		await login(values).unwrap()
+	}
+
+	return (
+		<Formik
+			initialValues={{
```
Comment (Ruslan1516) [inline src/components/LoginForm/LoginForm.tsx:46]: and this

**Review by Ruslan1516 — APPROVED** [review]

**Pull request merged**